File: obsPlugin/src/main/resources/static/js/cuesheet/datamodel/markerData.js
/**
 * Created by Broadcast Media Solutions on 05/11/2015.
 */

(function (AV) {
    var MarkerData;

    AV.ns("AV.obsPlugin.datamodel");

    MarkerData = function (data) {
        this.start = "";
        this.duration = "";
        this.comment = "";

        if (data) {
            this.start = data.start;
            this.duration = data.duration;
            this.comment = data.comment;
        }

    };


    function toJs()
    {
        return JSON.stringify(this);
    }

    AV.obsPlugin.datamodel.MarkerData = MarkerData;
}(AV));

File: obsPlugin/src/main/java/com/avid/central/obsplugin/datamodel/MarkerData.java
package com.avid.central.obsplugin.datamodel;

import javax.xml.bind.annotation.XmlElement;

/**
 * Created by Broadcast Media Solutions on 10/11/2015.
 */
public class MarkerData {
    @XmlElement(required = true)
    public String Start;
    @XmlElement(required = true)
    public String Duration;
    @XmlElement(required = true)
    public String Comment;

    public String toString()
    {
        StringBuilder sb = new StringBuilder(Start);
        sb.append(",");
        sb.append(Duration);
        sb.append(",\"");
        sb.append(Comment);
        sb.append("\"");
        return sb.toString();
    }
}
